File: magnet-processor/src/test/resources/MagnetProcessorTest/Limit_ScopingDirect_GeneratesGetter/UnderTest.java
package app;

import magnet.Instance;
import magnet.Scope;
import magnet.Scoping;

@Instance(
    type = UnderTest.class,
    scoping = Scoping.DIRECT,
    limitedTo = "activity"
)
public class UnderTest {
    public UnderTest(Scope scope) {
    }
}

File: magnet-processor/src/test/java/UnderTest.kt
package app

import magnet.Instance
import magnet.Scope

@Instance(type = UnderTest::class)
class UnderTest constructor(
    private val parentScope: Scope,
    private val value: String = "default"
) {

    constructor(parentScope: Scope) : this(parentScope, "default")

}